src/main/java/com/google/devtools/build/lib/packages/Type.java

```diff
@@ -559,6 +560,15 @@ public Map<KeyT, ValueT> convert(Object x, Object what, Object context)
       return ImmutableMap.copyOf(result);
     }
 
+    @Override
+    public Map<KeyT, ValueT> concat(Iterable<Map<KeyT, ValueT>> iterable) {
+      LinkedHashMap<KeyT, ValueT> output = new LinkedHashMap<>();
```

Comment:
Presumably LinkedHashMap is used instead of ImmutableMap.Builder (at the cost of a copy) to avoid an exception when the keys intersect. A comment to this effect would be good.

Better still: use Dict.Builder, which avoids the exception, and will take advantage of the exception-free builder semantics (buildKeepingLast) when they become available.

Reply:
`buildKeepingLast` is in fact already available: https://cs.opensource.google/bazel/bazel/+/4eb62cd07393e495ab9ead49b771871d1050ee51:src/main/java/com/google/devtools/build/lib/actions/ActionEnvironment.java;l=111

Reply:
Updated to use `buildKeepingLast`, although from the docs:
> In the current implementation, all values associated with a given key are stored in the Builder object, even though only one of them will be used in the built map. If there can be many repeated keys, it may be more space-efficient to use a [LinkedHashMap](https://docs.oracle.com/javase/9/docs/api/java/util/LinkedHashMap.html?is-external=true) and [ImmutableMap.copyOf(Map)](https://guava.dev/releases/31.1-jre/api/docs/com/google/common/collect/ImmutableMap.html#copyOf(java.util.Map)) rather than ImmutableMap.Builder.